flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/utils/ExecNodeMetadataUtilTest.java

```diff
@@ -338,11 +455,88 @@ protected DummyNodeNoAnnotation(
         @JsonCreator
         protected DummyNodeBothAnnotations(
                 ExecNodeContext context,
-                ReadableConfig config,
+                ReadableConfig persistedConfig,
+                List<InputProperty> properties,
+                LogicalType outputType,
+                String description) {
+            super(10, context, persistedConfig, properties, outputType, description);
+        }
+
+        @Override
+        protected Transformation<RowData> translateToPlanInternal(
+                PlannerBase planner, ExecNodeConfig config) {
+            return null;
+        }
+    }
+
+    @ExecNodeMetadata(
+            name = "dummy-node-duplicate-consumedOptions",
+            version = 3,
+            consumedOptions = {"option1", "option2", "option3", "option2"},
+            minPlanVersion = FlinkVersion.v1_15,
+            minStateVersion = FlinkVersion.v1_15)
+    private static class DummyNodeDuplicateConsumedOptions extends ExecNodeBase<RowData> {
+
+        @JsonCreator
+        protected DummyNodeDuplicateConsumedOptions(
+                ExecNodeContext context,
+                ReadableConfig persistedConfig,
+                List<InputProperty> properties,
+                LogicalType outputType,
+                String description) {
+            super(10, context, persistedConfig, properties, outputType, description);
+        }
+
+        @Override
+        protected Transformation<RowData> translateToPlanInternal(
+                PlannerBase planner, ExecNodeConfig config) {
+            return null;
+        }
+    }
+
+    @ExecNodeMetadata(
+            name = "dummy-node-duplicate-deprecated-keys-consumedOptions",
+            version = 3,
+            consumedOptions = {"option1", "option2", "option3", "option11"},
+            minPlanVersion = FlinkVersion.v1_15,
+            minStateVersion = FlinkVersion.v1_15)
+    private static class DummyNodeDuplicateDeprecatedKeysConsumedOptions
+            extends ExecNodeBase<RowData> {
+
+        @JsonCreator
+        protected DummyNodeDuplicateDeprecatedKeysConsumedOptions(
+                ExecNodeContext context,
+                ReadableConfig persistedConfig,
+                List<InputProperty> properties,
+                LogicalType outputType,
+                String description) {
+            super(10, context, persistedConfig, properties, outputType, description);
+        }
+
+        @Override
+        protected Transformation<RowData> translateToPlanInternal(
```

Comment:
create a base class for all these dummies, in case someone adds a method to `ExecNodeBase` they only need to update one location.

Reply:
Thx, would also clean up some code.